server/tests/routes/group.test.js

```diff
@@ -139,28 +163,63 @@ describe('## Group API', () => {
       });
 
       describe('# DELETE /api/v1/group/:groupName/user/:userId', () => {
-        const data = {
-          userId: group.users[0]
-        };
+        const userObj = fixtures.user();
+        let user;
+        let group;
+
+        beforeEach((done) => {
+          build('user', userObj)
+            .then((u) => {
+              user = u;
+              const groupObj= fixtures.group({ users: [user.id] });
+              build('group', groupObj)
+                .then((g) => {
+                  group = g;
+                  userService.addGroupByUserId(user.id, group.name)
```

Comment:
Same comment as above, it looks like you could reduce some of the pyramid here. You save the user into a variable in the parent scope, which means you don't have to keep nesting the subsequent function calls.